vue/src/components/JoistUpload.vue

```diff
@@ -105,38 +106,44 @@ export default class JoistUpload extends Vue {
       .on('httpUploadProgress', (evt) => {
         const s3Progress = evt.loaded / evt.total;
         // s3Progress only spans the total fileProgress range [0.1, 0.9)
-        this.fileProgress = OVERHEAD_PERCENT / 2 + s3Progress * (1 - OVERHEAD_PERCENT);
+        onProgress(OVERHEAD_PERCENT / 2 + s3Progress * (1 - OVERHEAD_PERCENT));
       })
       .promise();
 
     await http.request({
       method: 'post',
-      url: 'finalize-upload/',
+      url: 'joist/finalize-upload/',
       data: {
         name: initUpload.objectKey,
       },
     });
-    this.fileProgress = 1;
+
+    onProgress(1);
+
+    return initUpload.objectKey;
   }
 
   public reset() {
     this.fileInput.value = '';
     this.submitting = false;
-    this.currentUploadNum = 0;
-    this.fileProgress = 0;
   }
 
   @Emit('complete')
   public async submit() {
     this.submitting = true;
 
-    this.currentUploadNum = 0;
-    for (const file of this.files) { // eslint-disable-line no-restricted-syntax
-      await this.uploadFile(file); // eslint-disable-line no-await-in-loop
+    const uploaded = this.files.map(file => JoistUpload.uploadFile(file.file, (p) => {
+      file.progress = p; // eslint-disable-line no-param-reassign
+    }));
+    const names = await Promise.all(uploaded);
```

Comment:
Do you think it's safe to let an arbitrary number of files upload in parallel? I had it run serially in case there was an interruption, since we'd know the point at which it got interrupted.

I don't feel strongly, just wondering.

Reply:
atm it doesn't matter much since the `multiple` flag is not set in the file field. Thus, the user can select a single field anyhow.